iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/atomic_relocatable_ptr.hpp

```diff
@@ -21,45 +23,57 @@
 
 namespace iox
 {
-///@brief minimalistic relocatable pointer that can be written and read atomically
-/// and can be stored safely in shared memory.
-/// As the basic relocatable_ptr, it must point to something in the same shared memory segment as itself
-/// since the internally used offset must be an invariant different across adress spaces.
-/// Rationale: the default relocatable_ptr cannot be used in an atomic since the copy ctor is nontrivial.
+namespace rp
+{
+/// @brief minimalistic relocatable pointer that can be written and read atomically and can be stored safely in shared
+/// memory.
+/// As the basic RelocatablePointer, it must point to something in the same shared memory segment as itself since the
+/// internally used offset must be an invariant different across adress spaces. Rationale: the default
+/// RelocatablePointer cannot be used in an atomic since the copy ctor is nontrivial.
 template <typename T>
 class atomic_relocatable_ptr
 {
   public:
     using offset_t = std::ptrdiff_t;
     static constexpr offset_t NULL_POINTER_OFFSET = std::numeric_limits<offset_t>::max();
 
-    atomic_relocatable_ptr(const T* ptr = nullptr);
+    /// @brief creates atomic_relocatable_ptr pointing to ptr
```

Comment:
It creates an atomic_relocatable_ptr that points to the same pointee as ptr **from ptr**, but not to ptr itself.